networkit/cpp/graph/GraphTools.cpp

```diff
@@ -504,5 +504,43 @@ void sortEdgesByWeight(Graph &G, bool decreasing) {
         });
 }
 
+std::vector<node> topologySort(const Graph &G) {
+    if (!G.isDirected()) {
+        throw std::runtime_error(
+            "Topological sort is defined for directed graphs only.");
+    }
+
+    enum nodeMark { NONE, TEMP, PERM };
+    std::vector<nodeMark> topSortMark(G.numberOfNodes(), nodeMark::NONE);
+    std::vector<node> result(G.numberOfNodes(), 0);
+    count current = G.numberOfNodes() - 1;
+    std::function<void(node)> exploreNode;
+
+    exploreNode = [&](node u) {
```

Comment:
This still uses `std::function` (which has a high overhead) even though that is not necessary:
```suggestion
    auto exploreNode = [&](node u) {
```

Reply:
Ah wait, you're doing this to enable the recursion. It's probably better to use a normal function in this case (and not a lambda).